packages/blockchain-sdk-evm/src/auth/evm-ed25519-authentication-facade-factory.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  AuthenticationFacade,
+  AuthenticationFacadeFactory,
+  Authenticator,
+  DefaultTokenGenerator,
+  Ed25519TokenBodyParser,
+  TokenParser,
+} from '@dialectlabs/sdk';
+
+import { EvmEd25519TokenValidator } from './evm-ed25519-token-validator';
+import type { EvmEd25519TokenSigner } from './evm-ed25519-token-signer';
+
+export class EvmEd25519AuthenticationFacadeFactory extends AuthenticationFacadeFactory {
+  constructor(private readonly tokenSigner: EvmEd25519TokenSigner) {
+    super();
+  }
+
+  static createAuthenticator(): Authenticator {
+    return new Authenticator(
+      new TokenParser(new Ed25519TokenBodyParser()),
+      new EvmEd25519TokenValidator(),
+    );
+  }
+
+  get(): AuthenticationFacade {
+    return new AuthenticationFacade(
+      this.tokenSigner,
+      new DefaultTokenGenerator(this.tokenSigner),
+      EvmEd25519AuthenticationFacadeFactory.createAuthenticator(),
+    );
+  }
+}
```

Comment:
this is def a proven to work implementation suitable for our backend. But EVM has it's own specification describing how to implement web2 authentication. I'm not insisting to change it rn, but still good to consider and be aware of pros and cons in comparison to the current implementation
https://eips.ethereum.org/EIPS/eip-4361

@tsmbl wdyt?

Reply:
Thanks for bringing this spec up! I def think this paper makes a lot of sense so I would def consider it to be aware about pros and cons in comparison to the current implementation, also there's super-useful security section. As I understand this EIP is in still in review stage, so I think we need to study it and be aware, but probably hold spec implementation for now
I personally think we will need to revisit token structure for all chains at some point, not only for evm
1. enable human readable data in message signing (as I remember it was discussed some time ago)
2. take into account federated environments / sso with multiple backends owned by different parties
3. smth else?

wdyt? @zhelezkov @AlekseiPetrovDev

Reply:
sg for me